File: Hackyto/Hackyto/Helpers/ColorFactory.swift
//
//  ColorFactory.swift
//  Hackyto
//
//  Created by Ernesto Torres on 12/21/15.
//  Copyright © 2015 ehtd. All rights reserved.
//

import UIKit

class ColorFactory: NSObject {

    class func darkColor() -> UIColor {
        return UIColor(red: 22/255.0, green: 22/255.0, blue: 23/255.0, alpha: 1.0)
    }

    class func darkGrayColor() -> UIColor {
        return UIColor(red: 60/255.0, green: 60/255.0, blue: 60/255.0, alpha: 1.0)
    }

    class func lightColor() -> UIColor {
        return UIColor.white
    }

    class func blueColor() -> UIColor {
        return UIColor(red: 41/255, green: 128/255, blue: 185/255, alpha: 1.0)
    }

    class func turquoiseColor() -> UIColor {
        return UIColor(red: 26/255, green: 188/255, blue: 156/255, alpha: 1.0)
    }

    class func emeraldColor() -> UIColor {
        return UIColor(red: 46/255, green: 204/255, blue: 113/255, alpha: 1.0)
    }

    class func amethystColor() -> UIColor {
        return UIColor(red: 155/255, green: 89/255, blue: 182/255, alpha: 1.0)
    }

    class func sunFlowerColor() -> UIColor {
        return UIColor(red: 241/255, green: 196/255, blue: 15/255, alpha: 1.0)
    }

    class func carrotColor() -> UIColor {
        return UIColor(red: 230/255, green: 126/255, blue: 34/255, alpha: 1.0)
    }

    class func alizarinColor() -> UIColor {
        return UIColor(red: 231/255, green: 76/255, blue: 60/255, alpha: 1.0)
    }

    class func wisteriaColor() -> UIColor {
        return UIColor(red: 142/255, green: 68/255, blue: 173/255, alpha: 1.0)
    }

    class func orangeColor() -> UIColor {
        return UIColor(red: 243/255, green: 156/255, blue: 18/255, alpha: 1.0)
    }

    class func pumpkinColor() -> UIColor {
        return UIColor(red: 211/255, green: 84/255, blue: 0/255, alpha: 1.0)
    }

    class func belizeHoleColor() -> UIColor {
        return UIColor(red: 41/255, green: 128/255, blue: 185/255, alpha: 1.0)
    }

    class func colorFromNumber(_ number: Int) -> UIColor {
        let colors = [
            belizeHoleColor(),
            turquoiseColor(),
            orangeColor(),
            alizarinColor(),
            emeraldColor(),
            sunFlowerColor(),
            amethystColor(),
            carrotColor(),
            pumpkinColor(),
            wisteriaColor()
            ]

        let index = number % 10
        return colors[index]
    }
}

File: Android/Hackyto/sharedLib/src/main/java/io/ernesto/hn/ColorFactory.java
package io.ernesto.hn;

/**
 * Created by ernesto.torres on 8/11/17.
 */

public class ColorFactory {

    static public int darkGrayColor = 0x3C3C3C;
    static public int lightColor = 0xFFFFFF;
    static public int blueColor = 0x2980B9;
    static public int turquoiseColor = 0x1ABC9C;
    static public int emeraldColor = 0x2ECC71;
    static public int amethystColor = 0x9B59B6;
    static public int sunFlowerColor = 0xF1C40F;
    static public int carrotColor = 0xE67E22;
    static public int alizarinColor = 0xE74C3C;
    static public int wisteriaColor = 0x8E44AE;
    static public int orangeColor = 0xF39C12;
    static public int pumpkinColor = 0xD35400;
    static public int belizeHoleColor = 0x2980B9;

    public static int colorFromNumber(int n) {
        if (n < 0) { return lightColor; }

        int[] colors = new int[]{
                belizeHoleColor,
                turquoiseColor,
                orangeColor,
                alizarinColor,
                emeraldColor,
                sunFlowerColor,
                amethystColor,
                carrotColor,
                pumpkinColor,
                wisteriaColor};
        int index = n % colors.length;

        return colors[index];
    }

    public static int appendAlpha(int n, int alpha) {
        if (alpha > 0xFF || alpha < 0) {
            alpha = 0xFF;
        }

        int shiftedColor = n << 8;
        return shiftedColor | alpha;
    }

    public static int prefixAlpha(int n, int alpha) {
        if (alpha > 0xFF || alpha < 0) {
            alpha = 0xFF;
        }

        int shifted = alpha << (4 * 6);
        return n | shifted;
    }
}
